src/sentry/runner/commands/createuser.py

```diff
@@ -37,27 +37,46 @@ def _get_superuser():
     return click.confirm("Should this user be a superuser?", default=False)
 
 
-def _set_user_permissions(user):
+def _set_superadmin(user):
+    """
+    superadmin role approximates superuser (model attribute) but leveraging
+    Sentry's role system.
+    """
     from sentry.models import UserRole, UserRoleUser
 
-    if click.confirm(
-        "Should this user have Super Admin role? (This grants them all permissions available)",
-        default=False,
-    ):
-        role = UserRole.objects.get(name="Super Admin")
-        UserRoleUser.objects.create(user=user, role=role)
+    role = UserRole.objects.get(name="Super Admin")
+    UserRoleUser.objects.create(user=user, role=role)
 
 
 @click.command()
 @click.option("--email")
 @click.option("--password")
-@click.option("--superuser/--no-superuser", default=None, is_flag=True)
+@click.option(
+    "--superuser/--no-superuser",
+    default=None,
+    is_flag=True,
+    help="Superusers have full access to Sentry, across all organizations.",
+)
+@click.option(
+    "--staff/--no-staff",
+    default=None,
+    is_flag=True,
+    help="Staff users have access to Django backend.",
+)
 @click.option("--no-password", default=False, is_flag=True)
 @click.option("--no-input", default=False, is_flag=True)
-@click.option("--force-update", default=False, is_flag=True)
+@click.option(
+    "--force-update", default=False, is_flag=True, help="If true, will update existing users."
+)
 @configuration
-def createuser(email, password, superuser, no_password, no_input, force_update):
+def createuser(email, password, superuser, staff, no_password, no_input, force_update):
     "Create a new user."
+
+    from django.conf import settings
+
+    if not (settings.SENTRY_SELF_HOSTED or settings.SENTRY_SINGLE_ORGANIZATION):
+        return
```

Comment:
@rgibert Just want to understand the context on this change. At the moment, this is causing `createuser` to fail on `getsentry`.

Reply:
@wedamija @rgibert Did this get resolved somehow? Link?

Reply:
https://github.com/getsentry/sentry/pull/37630 I think?

Reply:
[Yup](https://github.com/getsentry/sentry/pull/32802#issuecomment-1215143946). 👍